default/AI/AIstate.py

```diff
@@ -662,6 +661,14 @@ def rate_fleet(self, fleet_id, enemy_stats=None, ship_info=None):
                 shields = ship.currentMeterValue(fo.meterType.shield)
                 stats['structure'] = structure
                 stats['shields'] = shields
+                stats['attack'] = 0
+                stats['attacks'] = {}
+                design = fo.getShipDesign(design_id)
+                for partname in design.parts:
```

Comment:
Nope, the bug occurred in a default quickstart game. I switched on auto-advance turns, built a super-tester takeover and just let the game run, for ~60 turns. By that time I got frequent chat error messages from AIs 1 and 3, checked the logs and there they were, in abundance.

So whatever is going on here has nothing to do with multiplayer/moderator games. I've also found a lot of error messages that a certain ship design could not be found, which occurred somewhere during effect processing. Looks as if some ships fly around which don't have a valid design assigned...